zerver/lib/test_classes.py

```diff
@@ -514,10 +514,13 @@ def send_and_test_private_message(self, fixture_name, expected_subject=None,
 
         return msg
 
-    def build_webhook_url(self):
-        # type: () -> Text
+    def build_webhook_url(self, **kwargs):
+        # type: (**Any) -> Text
         api_key = self.get_api_key(self.TEST_USER_EMAIL)
-        return self.URL_TEMPLATE.format(stream=self.STREAM_NAME, api_key=api_key)
+        url = self.URL_TEMPLATE.format(stream=self.STREAM_NAME, api_key=api_key)
+        if kwargs.get("branches") is not None:
```

Comment:
btw - your implementation does not handle `?` at the beginning of params section right?